python_modules/dagster/dagster/core/storage/schedules/migration.py

```diff
@@ -1,4 +1,57 @@
 from typing import Callable, Mapping
 
-REQUIRED_SCHEDULE_DATA_MIGRATIONS: Mapping[str, Callable] = {}
-OPTIONAL_SCHEDULE_DATA_MIGRATIONS: Mapping[str, Callable] = {}
+from tqdm import tqdm
+
+from ..schedules.schema import JobTickTable
+
+SCHEDULE_JOBS_SELECTOR_ID = "schedule_jobs_selector_id"
+SCHEDULE_TICKS_SELECTOR_ID = "schedule_ticks_selector_id"
+
+REQUIRED_SCHEDULE_DATA_MIGRATIONS: Mapping[str, Callable] = {
+    SCHEDULE_JOBS_SELECTOR_ID: lambda: add_selector_id_to_jobs_table,
+}
+OPTIONAL_SCHEDULE_DATA_MIGRATIONS: Mapping[str, Callable] = {
+    SCHEDULE_TICKS_SELECTOR_ID: lambda: add_selector_id_to_ticks_table,
+}
+
+
+def add_selector_id_to_jobs_table(storage, print_fn=None):
+    """
+    Utility method that calculates the selector_id for each stored instigator state, and writes
+    it to the jobs table.
+    """
+
+    if print_fn:
+        print_fn("Querying storage.")
+
+    instigator_states = storage.all_instigator_state()
+    for state in tqdm(instigator_states):
+        # should write the selector_id to the jobs table
+        storage.update_instigator_state(state)
```

Comment:
does this update any timestamps? That's probably fine as long as it doesn't change the start timestamp or anything

Reply:
it doesn't but it probably should... good catch, I'll rewrite the migration and maybe in a separate diff will change writing the update_timestamp